app/constants/errorTypes.js

```diff
@@ -0,0 +1,8 @@
+export default {
+  networkError: 'NETWORK_ERROR',
+  internalError: 'INTERNAL_ERROR',
+  serverError: 'SERVER_ERROR',
+  validateError: 'VALIDATE_ERROR',
+  userError: 'USER_ERROR',
+  miscError: 'MISCELLEANOUS_ERROR',
+};
```

Comment:
Changes:
- Subdivide errors into categories. 
- Specialise errors.
- Name errors like constants.


```suggestion
export default {
  AUTH: {
    LOGIN: {
      INVALID_PHONE: "INVALID_PHONE",
      MISSING_PHONE: "MISSING_PHONE",
      INVALID_OTP: "INVALID_OTP",
      MISSING_OTP: "MISSING_OTP",
    },
    NETWORK_ERROR: 'NETWORK_ERROR',
    SERVER_ERROR: 'SERVER_ERROR',
  }
};
```